src/main/java/kitchenpos/order/application/OrderService.java

```diff
@@ -0,0 +1,102 @@
+package kitchenpos.order.application;
+
+import kitchenpos.menu.domain.MenuRepository;
+import kitchenpos.order.domain.*;
+import kitchenpos.order.dto.OrderResponse;
+import kitchenpos.order.dto.OrderStatusChangeRequest;
+import kitchenpos.ordertable.domain.OrderTable;
+import kitchenpos.order.dto.OrderCreateRequest;
+import kitchenpos.ordertable.domain.OrderTableRepository;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.stream.Collectors;
+
+@Service
+@Transactional
+public class OrderService {
+    private final MenuRepository menuRepository;
+    private final OrderRepository orderRepository;
+    private final OrderTableRepository orderTableRepository;
+
+    public OrderService(
+            MenuRepository menuRepository, OrderRepository orderRepository,
+            OrderTableRepository orderTableRepository) {
+        this.menuRepository = menuRepository;
+        this.orderRepository = orderRepository;
+        this.orderTableRepository = orderTableRepository;
+    }
+
+    public OrderResponse create(OrderCreateRequest orderCreateRequest) {
+        OrderLineItems orderLineItems = generateOrderLineItems(orderCreateRequest);
+
+        OrderTable orderTable = generateOrderTable(orderCreateRequest);
+
+        Order savedOrder = generateOrder(orderTable, orderLineItems);
+
+        return OrderResponse.of(savedOrder);
+    }
+
+    private OrderTable generateOrderTable(OrderCreateRequest orderCreateRequest) {
+        OrderTable orderTable = orderTableRepository.findById(orderCreateRequest.getOrderTableId())
+                .orElseThrow(NoSuchElementException::new);
+
+        if (orderTable.isEmpty()) {
+            throw new IllegalArgumentException();
+        }
+        return orderTable;
+    }
+
+    private OrderLineItems generateOrderLineItems(OrderCreateRequest orderCreateRequest) {
+        List<OrderLineItem> orderLineItemList = orderCreateRequest.getOrderLineItems()
+                .stream()
+                .map(orderLineItemRequest -> new OrderLineItem(
+                        orderLineItemRequest.getMenuId(),
+                        orderLineItemRequest.getQuantity()
+                )).collect(Collectors.toList());
+
+        OrderLineItems orderLineItems = new OrderLineItems(orderLineItemList);
```

Comment:
리스트 생성과 동시에 일급 컬렉션으로 반드신다면 아래와 같이 표현해 볼 수 있을 것 같습니다 😄 
```suggestion
         orderCreateRequest.getOrderLineItems()
                .stream()
                .map(orderLineItemRequest -> new OrderLineItem(
                        orderLineItemRequest.getMenuId(),
                        orderLineItemRequest.getQuantity()
                )).collect(Collectors.collectingAndThen(Collectors.toList(),
                        OrderLineItems::new));
```

이 부분은 참고만 부탁 드리겠습니다 🙇

Reply:
오 이런 방법도 있었네요! 감사합니다!